code/modules/mining/machine_processing.dm

```diff
@@ -140,13 +164,28 @@
 	if(!materials.has_space(material_amount))
 		unload_mineral(O)
 	else
+		points += O.points * O.amount
 		materials.insert_item(O)
 		qdel(O)
 		if(CONSOLE)
 			CONSOLE.updateUsrDialog()
 
 /obj/machinery/mineral/processing_unit/proc/get_machine_data()
 	var/dat = "<b>Smelter control console</b><br><br>"
+
+	//On or off
+	dat += "Machine is currently "
+	if (on)
+		dat += "<A href='?src=[REF(CONSOLE)];set_on=off'>On</A> "
+	else
+		dat += "<A href='?src=[REF(CONSOLE)];set_on=on'>Off</A> "
```

Comment:
```suggestion
	dat += "Machine is currently [ on ? "<A href='?src=[REF(CONSOLE)];set_on=off'>On</A> " : "<A href='?src=[REF(CONSOLE)];set_on=on'>Off</A> "]
```

Reply:
Actually why isn't a toggle? The function itself should check for on var and act on it, not aact on some href.

Reply:
As in move the on/off change onto the machine instead of the console?

Reply:
Yes, the function itself should be a toggle, that checks if the machine is on or off